src/index.d.ts

```diff
@@ -0,0 +1,280 @@
+import { ReactElement, ComponentType } from 'react'
+import { ReactTestInstance, ReactTestRenderer, act } from 'react-test-renderer'
+
+// EVENTS
+// ------
+
+// TODO: type second parameter
+export declare class NativeEvent {
+  constructor(typeArg: string, event?: any)
+}
+
+export declare function getEventHandlerName(key: string): string
+
+export interface FireEventFn {
+  (element: FiberRoot, event: NativeEvent): any
+  focus(element: FiberRoot, init?: any): any
+  blur(element: FiberRoot, init?: any): any
+  change(element: FiberRoot, init?: any): any
+  changeText(element: FiberRoot, value: string): any
+  contentSizeChange(element: FiberRoot, init?: any): any
+  endEditing(element: FiberRoot, init?: any): any
+  keyPress(element: FiberRoot, init?: any): any
+  submitEditing(element: FiberRoot, init?: any): any
+  layout(element: FiberRoot, init?: any): any
+  selectionChange(element: FiberRoot, init?: any): any
+  longPress(element: FiberRoot, init?: any): any
+  press(element: FiberRoot, init?: any): any
+  pressIn(element: FiberRoot, init?: any): any
+  pressOut(element: FiberRoot, init?: any): any
+  momentumScrollBegin(element: FiberRoot, init?: any): any
+  momentumScrollEnd(element: FiberRoot, init?: any): any
+  scroll(element: FiberRoot, init?: any): any
+  scrollBeginDrag(element: FiberRoot, init?: any): any
+  scrollEndDrag(element: FiberRoot, init?: any): any
+  load(element: FiberRoot, init?: any): any
+  error(element: FiberRoot, init?: any): any
+  progress(element: FiberRoot, init?: any): any
+}
+export declare const fireEvent: FireEventFn
+
+// GET NODE TEXT
+// -------------
+
+export declare function getNodeText(node: FiberRoot): string
+
+// GET QUERIES FOR ELEMENT
+// -----------------------
+
+export declare function getQueriesForElement<T = Queries>(element: ReactElement, queries?: T): BoundQueries<T>
+export declare function within<T = Queries>(element: ReactElement, queries?: T): BoundQueries<T>
+
+// PREETY PRINT
+// ------------
+
+// TODO: options are pretty-format options
+export declare function prettyPrint(element: ReactTestRenderer | FiberRoot, maxLength?: number, options?: any): string
```

Comment:
You can `import { OptionsReceived } from 'pretty-format'` to get the options.

Also, technically `element` can be anything. I would at least make it `ReactTestRenderer | FiberRoot | string